thirty-three-across/thirty-three-across-htb.js

```diff
@@ -56,40 +57,176 @@ function ThirtyThreeAcrossHtb(configs) {
-                },
+    function _getBoundingBox(element, minSize) {
+        var w = minSize.w;
+        var h = minSize.h;
+        var rect = Utilities.mergeObjects({
+            width: 0,
+            height: 0,
+            left: 0,
+            top: 0,
+            right: 0,
+            bottom: 0
+        }, element.getBoundingClientRect());
+
+        if ((rect.width === 0 || rect.height === 0) && w && h) {
+            rect.width = w;
+            rect.height = h;
+            rect.right = rect.left + w;
+            rect.bottom = rect.top + h;
+        }
+
+        return rect;
+    }
+
+    function _getIntersectionOfRects(rects) {
+        var bbox = {
+            left: rects[0].left,
+            right: rects[0].right,
+            top: rects[0].top,
+            bottom: rects[0].bottom
+        };
+
+        for (var i = 1; i < rects.length; ++i) {
+            bbox.left = Math.max(bbox.left, rects[i].left);
+            bbox.right = Math.min(bbox.right, rects[i].right);
+
+            if (bbox.left >= bbox.right) {
+                return null;
+            }
+
+            bbox.top = Math.max(bbox.top, rects[i].top);
+            bbox.bottom = Math.min(bbox.bottom, rects[i].bottom);
+
+            if (bbox.top >= bbox.bottom) {
+                return null;
+            }
+        }
+
+        bbox.width = bbox.right - bbox.left;
+        bbox.height = bbox.bottom - bbox.top;
+
+        return bbox;
+    }
+
+    function _getPercentInView(element, topWin, minSize) {
+        var elementBoundingBox = _getBoundingBox(element, minSize);
+
+        // Obtain the intersection of the element and the viewport
+        var elementInViewBoundingBox = _getIntersectionOfRects([
+            {
+                left: 0,
+                top: 0,
+                right: topWin.innerWidth,
+                bottom: topWin.innerHeight
+            },
+            elementBoundingBox
+        ]);
+
+        var elementInViewArea;
+        var elementTotalArea;
+
+        if (elementInViewBoundingBox !== null) {
+            // Some or all of the element is in view
+            elementInViewArea = elementInViewBoundingBox.width * elementInViewBoundingBox.height;
+            elementTotalArea = elementBoundingBox.width * elementBoundingBox.height;
+
+            return (elementInViewArea / elementTotalArea) * 100;
+        }
+
+        // No overlap between element and the viewport; therefore, the element lies completely out of view
+        return 0;
+    }
+
+    function _isIframe() {
+        /*
+         * NOTE: this function should doing the following, otherwise revisit this code
+         *  try {
+         *      return window.top !== window.self;
+         *  } catch (e) {
+         *      return true;
+         *  }
+         */
+        return !Browser.isTopFrame();
+    }
+
+    function _getAdSlotHTMLElement(parcel) {
+        var slotElementId = null;
+
+        // NOTE: viewability feature supports the DFP tag only for now
```

Comment:
If it's not GPT then do we return `nm`?

Reply:
yes. currently thats what prebid does, so i guess IX adapter should follow it too